tools/generators/migrate-v8-pkg/index.ts

```diff
@@ -33,18 +56,131 @@ function normalizeOptions(tree: Tree, options: MigrateV8PkgGeneratorSchema) {
+  if (shouldValidateNameInput()) {
+    throw new Error(`--name cannot be empty. Please provide name of the package.`);
+  }
+
+  if (newSchema.name) {
+    const projectNames = newSchema.name.split(',').filter(Boolean);
+
+    projectNames.forEach(projectName => {
+      const projectConfig = readProjectConfiguration(tree, projectName);
+
+      if (!isPackageV8(tree, projectConfig)) {
+        throw new Error(
+          `${newSchema.name} is not v8 package. Make sure to run the migration on packages with version 8.x.x`,
+        );
+      }
+    });
+  }
+
+  return newSchema;
+}
+
+function isPackageV8(tree: Tree, project: ProjectConfiguration) {
+  const packageJson = readJson<PackageJson>(tree, joinPathFragments(project.root, 'package.json'));
+  return packageJson.version.startsWith('8.');
+}
+
+function isProjectMigrated<T extends ProjectConfiguration>(
+  tree: Tree,
+  project: T,
+): project is T & Required<Pick<ProjectConfiguration, 'tags' | 'sourceRoot'>> {
+  const hasTsSolutionConfigSetup = Array.isArray(
+    readJson<TsConfig>(tree, joinPathFragments(project.root, 'tsconfig.json')).references,
   );
+  // eslint-disable-next-line eqeqeq
+  return project.sourceRoot != null && Boolean(project.tags?.includes('v8')) && hasTsSolutionConfigSetup;
+}
+
+function getProjectMetadata(tree: Tree, project: ProjectConfiguration) {
+  return {
+    packageJson: readJson<PackageJson>(tree, joinPathFragments(project.root, 'package.json')),
+    eslintConfig: getEslint(),
+  };
+
+  function getEslint() {
+    const paths = {
+      json: joinPathFragments(project.root, '.eslintrc.json'),
+      js: joinPathFragments(project.root, '.eslintrc.js'),
+    };
+
+    if (tree.exists(paths.json)) {
+      const eslintConfig: Linter.Config = readJson<Record<string, unknown>>(tree, paths.json);
+      return eslintConfig;
+    }
+
+    if (tree.exists(paths.js)) {
+      const eslintConfigAST = getCjsConfigObjectAst(tree.read(paths.js, 'utf8') as string);
+      const extendsFieldValue = getASTconfigObjectProp(eslintConfigAST, 'extends');
+
+      const eslintConfig: Linter.Config = { extends: extendsFieldValue };
+
+      return eslintConfig;
+    }
+  }
+}
+
+function printStats(tree: Tree, options: MigrateV8PkgGeneratorSchema) {
+  type ProjectStats = ProjectConfiguration & {
+    projectName: string;
+    metadata: ReturnType<typeof getProjectMetadata>;
+  };
+  const allProjects = getProjects(tree);
+  const stats = {
+    migrated: [] as Array<ProjectStats>,
+    notMigrated: [] as Array<ProjectStats>,
+  };
+
+  allProjects.forEach((project, projectName) => {
+    if (!isPackageV8(tree, project)) {
+      return;
+    }
+
+    const metadata = getProjectMetadata(tree, project);
+
+    if (isProjectMigrated(tree, project)) {
+      stats.migrated.push({ projectName, metadata, ...project });
+
+      return;
+    }
+    stats.notMigrated.push({ projectName, metadata, ...project });
+  });
+
+  function printProjectInfo(projectStat: ProjectStats) {
```

Comment:
Extreme nit, to the point that I feel pain writing this, but should we opt for something like `getProjectInfoMessage` or similar since this function prints nothing?